app/code/community/Bolt/Boltpay/controllers/OrderController.php

```diff
@@ -208,4 +208,59 @@ public function viewAction()
         }
     }
 
-}
\ No newline at end of file
+    /**
+     * Create ppcQuote and return token for the BoltProductCheckout
+     * @return Mage_Core_Controller_Response_Http
+     */
+    public function ppcAction()
+    {
+        if (!$this->getRequest()->isAjax()) {
+            Mage::throwException($this->boltHelper()->__("Bolt_Boltpay_OrderController::ppcAction called with a non AJAX call"));
+            return;
+        }
+        if (!$this->_validateFormKey()) {
+            Mage::throwException($this->boltHelper()->__("Bolt_Boltpay_OrderController::ppcAction form key is invalid"));
+            return;
+        }
+        $response = $this->_initProduct();
+        $token = $response->token;
+        /**
+         * Check product availability
+         */
+        if (!$token) {
+            Mage::throwException($this->boltHelper()->__("Bolt_Boltpay_OrderController::ppcAction product is empty"));
+            return;
+        }
+        $this->getResponse()->setHeader('Content-type', 'application/json', true);
+        return $this->getResponse()->setBody(Mage::helper('core')->jsonEncode($response));
+    }
+
+    /**
+     * 
```

Comment:
@oldDriver Can you add description for current method? (or maybe rename method)
It is a bit confusing method name _initProduct. As i see the method return a response with token and etc.